cloudinit/util.py

```diff
@@ -68,17 +68,27 @@
                    ['running-in-container'],
                    ['lxc-is-container'])
 
-PROC_CMDLINE = None
+_CACHED_RESPONSES = {}
 
-_LSB_RELEASE = {}
+def memoize():
+    """
+    Function used to cache responses from functions decorated with it.
+    """
+    global _CACHED_RESPONSES
+    def helpler(x):
+        if (f, x) not in _CASHED_RESPONSES:
+            _CACHED_RESPONSES[(f, x)] = f(x)
+        return _CACHED_RESPONSES[(f, x)]
 
```

Comment:
here's my `@memoize` function

i've decorated a few functions with it, that seemed like easy targets, *aside* from the ones using `global _DEFINED_UP_HERE`.